hail/python/hail/ir/matrix_ir.py

```diff
@@ -46,10 +66,17 @@ def __init__(self, child, pred):
         self.child = child
         self.pred = pred
 
+    @staticmethod
+    def new_block(i):
```

Comment:
sorry, something else that's confused me --

new_block conceptually indicates a different value IR group, right? In that case, it feels like  only value IRs should have a concept of a new block -- all relational IRs will define a new block for all value IR children. 

Can we remove this from table/matrix/blockmatrix?

Reply:
`new_block` conceptually indicates that there is some non-trivial control flow in the relationship between this node and its `i`th child. Or it would if I were being consistent and made loop bodies be new blocks. So it really just means "CSE won't add lets across this boundary". (Making loop bodies new blocks is technically correct for CSE to never add work to any control path, but heuristically we assume loops will run at least once.)

I considered doing it that way, but it feels to me like part of the structure of the IR, and should not be encoded differently in relational IR vs. value IR. If I remove it from BaseIR, then CSE will have to do a case on whether an IR is a value IR, which feels wrong to me.

It's not a big deal, and an easy change to make, if you want me to.

Reply:
ok, I'm fine leaving this on the relational nodes, but it should just be `return True` on the base classes, right?

Reply:
`TableIR`, `MatrixIR`, and `BlockMatrixIR` have overrides to `return True`, while `IR` overrides it to `return False`, with some subclasses like `If` overriding again. I think you're saying it would be simpler to define it as `True` in `BaseIR`? I can do that.

Reply:
er, right -- I mean delete all of the overrides on relational classes, e.g. https://github.com/hail-is/hail/pull/7000/files#diff-69e69044397dd5692bf5013604bdfadaR322

Reply:
Oh whoops! I didn't realize I had done that.

Reply:
thanks! sorry for the confusion